### PR TITLE
ENH: Added coverage option for `test` command

### PR DESCRIPTION
**Pull request by ganesh-k13**
### Changes
- ENH: Added coverage option for `test` command

### Testing
```
spin test --coverage -- -vv
```
Output:
```
...
Coverage HTML written to dir /home/ganesh/os/devpy/example_pkg/build/coverage/
...
```

```
~/os/devpy/example_pkg (coverage) » ls build/coverage                                                                                                                 130 ↵ ganesh@ganesh-MS-7B86
coverage_html.js  d_8d15ffec5afca27f___init___py.html  d_b2ad4cdda83343df_test_core_py.html  favicon_32.png  index.html  keybd_closed.png  keybd_open.png  status.json  style.css
```

### CODE REVIEW

**Comment by stefanv** [comment]
Thanks, Ganesh. I recently did a bunch of work on NumPy's commands, so will probably port that command over. I don't recall of it has a coverage option, but if not will integrate your changes from here. Thanks again!

File: spin/cmds/meson.py [hunk 1]
```diff
@@ -165,8 +165,13 @@ def _get_configured_command(command_name):
 
 @click.command()
 @click.argument("pytest_args", nargs=-1)
+@click.option(
+    "--coverage",
+    is_flag=True,
+    help="Report coverage of project code. HTML output goes under build/coverage",
```
Comment (stefanv) [inline spin/cmds/meson.py:171]:
```suggestion
    help="Generate coverage report of executed tests. HTML output goes under `build/coverage`.",
```

File: spin/cmds/meson.py [hunk 2]
```diff
@@ -236,6 +241,18 @@ def test(ctx, pytest_args):
             print("Stopping. Please investigate the build error.")
             sys.exit(1)
 
+    if coverage:
+        coverage_dir = os.path.join(os.getcwd(), "build/coverage/")
+        print(f"Removing `{coverage_dir}`")
+        if os.path.isdir(coverage_dir):
+            shutil.rmtree(coverage_dir)
```
Comment (stefanv) [inline spin/cmds/meson.py:248]:
```suggestion
        if os.path.isdir(coverage_dir):
            print(f"Removing `{coverage_dir}`")
            shutil.rmtree(coverage_dir)
```

File: spin/cmds/meson.py [hunk 3]
```diff
@@ -236,6 +241,18 @@ def test(ctx, pytest_args):
...
+        os.makedirs(coverage_dir)
+        pytest_args = [
+            *pytest_args,
+            f"--cov-report=html:{coverage_dir}",
+            f"--cov={package}",
+        ]
```
Comment (stefanv) [inline spin/cmds/meson.py:254]:
```suggestion
        pytest_args = pytest_args + [
            f"--cov-report=html:{coverage_dir}",
            f"--cov={package}",
        ]
```

**Comment by stefanv** [comment]
Thanks, Ganesh, much appreciated.

**Comment by ganesh-k13** [comment]
Thanks Stefan, I'll start work with NumPy now. I'll target the docs and CI (if any needed)

**Comment by stefanv** [comment]
Thanks very much.

**Pull request merged**